config/routes.rb

```diff
@@ -1,3 +1,18 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
+  devise_for :users, controllers: { omniauth_callbacks: 'callbacks' }
+  devise_scope :user do
+    authenticated :user do
+      root 'welcome#index', as: :authenticated_root
+    end
+    unauthenticated do
+      root 'devise/sessions#new', as: :unauthenticated_root
+    end
+  end
+
+  resources :subject, as: :subjects
```

Comment:
why not `resources :subjects`?

Reply:
Yes, you are right, just fixed that, it's a mistake when I was generating controller